src/openrct2/management/finance.c

```diff
@@ -76,6 +76,11 @@ void finance_payment(money32 amount, rct_expenditure_type type)
 	money32 new_money = cur_money - amount;
 
 	//overflow check
+	if (amount < 0 && cur_money >= INT_MAX + amount)
```

Comment:
because the game usually subtracts money from you more than adding moeny, financing generally will subtract whatever is input. if you are to remove a path piece, it'll do finance_payment(-10), for example.
Lets say INT_MAX was 100, and the cur_money is 95.
-10 is below 0, so that line is checked. 100 + - 10 is 90.
doing INT_MAX + 10 would cause an overflow, so you have to do the other way around and make sure that 95 isn't over 90.... 
honestly, it's weird as heck to explain. reading it myself is even hard to comprehend. but i've ran it through tests and it seems to work. i got the idea from the cheat_add_money. running the code through here shows that it's working correctly, http://www.tutorialspoint.com/compile_c_online.php?PID=0Bw_CjBb95KQMdUVtOG4zSEZDSWc